paddle/inference/example.cc

```diff
@@ -28,12 +30,29 @@ int main(int argc, char** argv) {
     exit(1);
   }
 
+  // 1. Define place, executor, scope
+  auto* place = new paddle::platform::CPUPlace();
+  paddle::framework::InitDevices();
+  paddle::framework::Executor* executor =
+      new paddle::framework::Executor(*place);
+  paddle::framework::Scope* scope = new paddle::framework::Scope();
+
   std::cout << "FLAGS_dirname: " << FLAGS_dirname << std::endl;
   std::string dirname = FLAGS_dirname;
 
-  paddle::InferenceEngine* engine = new paddle::InferenceEngine();
-  engine->LoadInferenceModel(dirname);
+  // 2. Initialize the inference program
+  paddle::framework::ProgramDesc* inference_program =
+      paddle::io::Load(*executor, *scope, dirname);
+
+  // 3. Optional: perform optimization on the inference_program
+
+  // 4. Get the feed_var_names and fetch_var_names
+  std::vector<std::string> feed_var_names =
```

Comment:
maybe change feed/fetch_var_names to feed/fetch_target_names to make them consistent with python side code.

Reply:
Sure, will modify. Thanks.

Reply:
I think it gets a bit confusing with feed_targets and fetch_targets. Maybe I will leave this for now.

Reply:
There are [feed_var_names](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/fluid/io.py#L238) and [feed_var_name](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/fluid/executor.py#L208) in python implementation. But there are not the same things: `feed_var_names` is the output's names of `feed_op` and `feed_var_name` is the input's names of `feed_op`. So we should distinguish this. Now we use `feed_targets` and `feed_holder` respectively now. Do you have any other suggestion?